File: systems-for-design-and-implementation/labs/lab3/LibraryPersistance/src/library/persistance/repository/Log.scala
package library.persistance.repository

import java.io.IOException
import java.util.logging.{FileHandler, Logger, SimpleFormatter}

object Log
{
    private val logger: Logger = Logger.getLogger("log")
    private var added: Boolean = false

    def get: Logger =
    {
        if (!added)
        {
            try
            {
                val file: FileHandler = new FileHandler("sdi.log")
                file.setFormatter(new SimpleFormatter)
                logger.addHandler(file)
            }
            catch
            {
                case e: IOException =>
                {
                    logger.warning("Failed to add file handle for log")
                }
            }
            added = true
        }
        logger
    }
}

File: systems-for-design-and-implementation/labs/exam/TicTacToePersistance/src/tictactoe/persistance/repository/Log.java
package tictactoe.persistance.repository;

import java.io.IOException;
import java.util.logging.FileHandler;
import java.util.logging.Logger;
import java.util.logging.SimpleFormatter;

// TODO add log to file from proprieties file
public class Log
{
    private static Logger  logger = Logger.getLogger("log");
    private static boolean added  = false;

    /**
     * Getter for property 'logger'.
     *
     * @return Value for property 'logger'.
     */
    public static Logger get()
    {
        if (!added)
        {
            try
            {
                FileHandler file = new FileHandler("sdi.log");
                file.setFormatter(new SimpleFormatter());
                logger.addHandler(file);
            }
            catch (IOException e)
            {
                logger.warning("Failed to add file handle for log");
            }
            added = true;
        }
        return logger;
    }
}
